operator/csv-generator/csv-generator.go

```diff
@@ -0,0 +1,224 @@
+			operatorCSV.Spec.CustomResourceDefinitions.Owned[i].Description = "Observatorium"
+			operatorCSV.Spec.CustomResourceDefinitions.Owned[i].DisplayName = "Observatorium"
+		}
+	}
+
+	// Re-serialize deployments and permissions into csv strategy.
+	operatorCSV.Spec.InstallStrategy.StrategySpec = strategySpec
+
+	operatorCSV.Annotations["containerImage"] = *operatorImage
+	for key, value := range userData.ExtraAnnotations {
+		operatorCSV.Annotations[key] = value
+	}
+
+	// Set correct csv versions and name
+	v := version.OperatorVersion{Version: *semverVersion}
+	operatorCSV.Spec.Version = v
+	operatorCSV.Name = "observatorium-operator.v" + *csvVersion
+	if *replacesCsvVersion != "" {
+		operatorCSV.Spec.Replaces = "observatorium-operator.v" + *replacesCsvVersion
+	} else {
+		operatorCSV.Spec.Replaces = ""
+	}
+
+	// Set api maturity
+	operatorCSV.Spec.Maturity = "alpha"
+
+	// Set links
+	operatorCSV.Spec.Links = []csvv1.AppLink{
+		{
+			Name: "Source Code",
+			URL:  "https://github.com/observatorium/configuration",
+		},
+	}
+
+	// Set Keywords
+	operatorCSV.Spec.Keywords = []string{
+		"observatorium",
+		"prometheus",
+		"thanos",
+	}
+
+	// Set Provider
+	operatorCSV.Spec.Provider = csvv1.AppLink{
+		Name: "Red Hat",
+	}
+
+	// Set Description
+	operatorCSV.Spec.Description = userData.Description
+	operatorCSV.Spec.DisplayName = "Observatorium Operator"
+
+	if userData.Maintainers != nil {
+		operatorCSV.Spec.Maintainers = []csvv1.Maintainer{}
+		for name, email := range userData.Maintainers {
+			operatorCSV.Spec.Maintainers = append(operatorCSV.Spec.Maintainers, csvv1.Maintainer{
+				Name:  name,
+				Email: email,
+			})
+		}
+	}
+
+	// Set Annotations
+	if *skipRange != "" {
+		operatorCSV.Annotations["olm.skipRange"] = *skipRange
+	}
+
+	// write CSV to out dir
+	writer := strings.Builder{}
+	MarshallObject(operatorCSV, &writer)
+	outputFilename := filepath.Join(*outputDir, finalizedCSVFilename())
+	err := ioutil.WriteFile(outputFilename, []byte(writer.String()), 0644)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("CSV written to %s\n", outputFilename)
+}
+
+func readFileOrPanic(filename string) []byte {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		panic(err)
+	}
+	return data
+}
+
+func readKeyValueMapFromFileOrPanic(filename string) map[string]string {
+	kvMap := make(map[string]string)
+	if err := json.Unmarshal(readFileOrPanic(filename), &kvMap); err != nil {
+		panic(err)
+	}
+	return kvMap
+}
+
+func main() {
+	flag.Parse()
+
+	if *csvVersion == "" {
+		log.Fatal("--csv-version is required")
+	} else if *operatorCSVTemplate == "" {
+		log.Fatal("--operator-csv-template-file is required")
+	} else if *operatorImage == "" {
+		log.Fatal("--operator-image is required")
+	} else if *outputDir == "" {
+		log.Fatal("--olm-bundle-directory is required")
+	}
+
+	var err error
+	// Set correct csv versions and name
+	semverVersion, err = semver.New(*csvVersion)
+	if err != nil {
+		panic(err)
```

Comment:
this should not panic. If someone happens to pass an invalid tag to the program, this is user error and should be handled appropriately

Reply:
removed all panics :)